pgmpy/factors/distributions/CustomDistribution.py

```diff
@@ -0,0 +1,509 @@
+
+        for var in variables:
+            if var not in self.variables:
+                raise ValueError("{var} not in scope.".format(var=var))
+
+        phi = self if inplace else self.copy()
+
+        all_var = [var for var in self.variables]
+        var_to_keep = [var for var in self.variables if var not in variables]
+        reordered_var_index = [all_var.index(var) for var in variables + var_to_keep]
+        pdf = phi._pdf
+
+        # The arguments need to be reordered because integrate.nquad
+        # integrates the first n-arguments of the function passed.
+
+        def reordered_pdf(*args):
+            # ordered_args restores the original order as it was in self.variables
+            ordered_args = [args[reordered_var_index.index(index_id)] for index_id in range(len(all_var))]
+            return pdf(*ordered_args)
+
+        def marginalized_pdf(*args):
+            return integrate.nquad(reordered_pdf, [[-np.inf, np.inf] for i in range(len(variables))], args=args)[0]
+
+        phi._pdf = marginalized_pdf
+        phi.variables = var_to_keep
+
+        if not inplace:
+            return phi
+        
+    def normalize(self, inplace=True):
+        """
+        Normalizes the pdf of the continuous distribution so that it
+        integrates to 1 over all the variables.
+
+        Parameters
+        ----------
+        inplace: boolean
+            If inplace=True it will modify the factor itself, else would return
+            a new factor.
+
+        Returns
+        -------
+        ContinuousFactor or None:
+             if inplace=True (default) returns None
+             if inplace=False returns a new ContinuousFactor instance.
+
+        Examples
+        --------
+        >>> from pgmpy.factors.distributions import CustomDistribution
+        >>> from scipy.stats import multivariate_normal
+        >>> normal_pdf_x2 = lambda x1, x2: 2 * multivariate_normal.pdf(
+        ...                     x=[x1, x2], mean=[0, 0], cov=[[1, 0], [0, 1]])
+        >>> normal_dist_x2 = CustomDistribution(variables=['x1', 'x2'],
+        ...                                     distribution=normal_pdf_x2)
+        >>> normal_dist_x2.assignment(1, 1)
+        0.117099663049
+        >>> normal_dist = normal_dist_x2.normalize(inplace=False))
+        >>> normal_dist.assignment(1, 1)
+        0.0585498315243
+        """
+        phi = self if inplace else self.copy()
+        pdf = self.pdf
+
+        pdf_mod = integrate.nquad(pdf, [[-np.inf, np.inf] for var in self.variables])[0]
+
+        phi._pdf = lambda *args: pdf(*args) / pdf_mod
+
+        if not inplace:
+            return phi
+
+    def _operate(self, other, operation, inplace=True):
+        """
+        Gives the ContinuousFactor operation (product or divide) with
+        the other factor.
+
+        Parameters
+        ----------
+        other: ContinuousFactor
+            The ContinuousFactor to be multiplied.
+
+        operation: String
```

Comment:
Should be just `str`. We try to use the exact strings that python's type function returns.